frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx

```diff
@@ -53,13 +58,37 @@ const LogsTabForPodName: React.FC<{ podName: string }> = ({ podName }) => {
   const [downloading, setDownloading] = React.useState(false);
   const [downloadError, setDownloadError] = React.useState<Error | undefined>();
   const { scrollToBottom, onMount, editorOptions } = useCodeEditorAsLogs();
+  const [isDownloadDropdownOpen, setIsDownloadDropdownOpen] = React.useState(false);
+  const [windowWidth, setWindowWidth] = React.useState(window.innerWidth);
+  const handleWindowResize = () => {
+    setWindowWidth(window.innerWidth);
+  };
+
+  React.useEffect(() => {
+    window.addEventListener('resize', handleWindowResize);
+    return () => {
+      window.removeEventListener('resize', handleWindowResize);
+    };
+  }, []);
```

Comment:
Couple things to note here.

1. If you're writing a listen for a `useEffect` you should always put it inside the `useEffect` or use a `useCallback`. Stable references to unloading event handlers are needed to actually unload the event handler -- your current usage won't break anything, but unless everyone who works on the dashboard is a react hook expert, this scenario will quickly fail if you try to do anything else with it
2. You should avoid writing several hyper dependant logic inline like this -- ideally create a common utility like `useWindowResize()` that creates the state, sets up the effect, and returns what is needed -- `windowWidth` for instance. I think for your use-case, knowing the window size is not needed by any caller, so you can start by returning a function `isSmallScreen` in an object so we can extend it as needed. For instance, `setWindowWidth` is something no one should ever call on their own -- the goal here is to make maintaining this code easier as the touch points are reduced -- eg. I won't need to know I shouldn't just call `setWindowWidth` whenever I want (which would break your usage the data)

Reply:
Created a utility. Just used `useCallback` to memoize `handleResize()`, not sure if I addressed the first comment here.